traffic_ops/testing/api/v4/cachegroups_test.go

```diff
@@ -835,3 +839,68 @@ func GetTestPaginationSupportCg(t *testing.T) {
 		t.Errorf("expected GET cachegroup to return an error for page is not a positive integer, actual error: %v - alerts: %+v", err, resp.Alerts)
 	}
 }
+
+func GetTestCacheGroupsByInvalidId(t *testing.T) {
+	opts := client.NewRequestOptions()
+	// Retrieve the CacheGroup to check CacheGroup name got updated
+	opts.QueryParameters.Set("id", "10000")
+	resp, _, _ := TOSession.GetCacheGroups(opts)
+	if len(resp.Response) > 0 {
+		t.Errorf("Expected 0 response, but got many %v", resp)
+	}
+}
+
+func GetTestCacheGroupsByInvalidType(t *testing.T) {
+	opts := client.NewRequestOptions()
+	// Retrieve the CacheGroup to check CacheGroup name got updated
+	opts.QueryParameters.Set("type", "10000")
+	resp, _, _ := TOSession.GetCacheGroups(opts)
+	if len(resp.Response) > 0 {
+		t.Errorf("Expected 0 response, but got many %v", resp)
+	}
+}
+
+func GetTestCacheGroupsByType(t *testing.T) {
+	if len(testData.CacheGroups) < 1 {
+		t.Fatal("Need at least one Cache Group to test updating Cache Groups")
+	}
+	firstCG := testData.CacheGroups[0]
+
+	if firstCG.Name == nil {
+		t.Fatal("Found Cache Group with null or undefined name in testing data")
+	}
+
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", *firstCG.Name)
+
+	resp, _, err := TOSession.GetCacheGroups(opts)
+	if err != nil {
+		t.Fatalf("cannot get Cache Group '%s': %v - alerts: %+v", *firstCG.Name, err, resp.Alerts)
+	}
+	if len(resp.Response) < 1 {
+		t.Fatalf("Expected exactly one Cache Group to exist with name '%s', but got: %d", *firstCG.Name, len(resp.Response))
```

Comment:
If the expectation is that "exactly one" exists, the condition should check for the length being exactly one or not, instead of merely less than one.

Reply:
updated the error message.